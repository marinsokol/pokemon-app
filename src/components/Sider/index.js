import React from 'react'
import { Link } from 'react-router-dom'
import { Layout, Menu, Icon } from 'antd'

const { Sider } = Layout
const { Item } = Menu

export default () => (
  <Sider collapsible>
    <Menu theme="dark" defaultSelectedKeys={['1']} mode="inline">
      <Item key="1">
        <Link to="/">
          <Icon type="heart-o" />
          <span>All pokemons</span>
        </Link>
      </Item>
      <Item key="2">
        <Link to="/favourites">
          <Icon type="star-o" />
          <span>Favourite pokemons</span>
        </Link>
      </Item>
    </Menu>
  </Sider>
)
