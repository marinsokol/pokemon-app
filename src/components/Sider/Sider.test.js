import React from 'react'
import { shallow } from 'enzyme'
import Sider from './index'

describe('Sider', () => {
  it('should be defined', () => {
    expect(Sider).toBeDefined()
  })

  it('should render correctly', () => {
    const tree = shallow(<Sider />)
    expect(tree).toMatchSnapshot()
  })

  it('should have 2 menu items', () => {
    const tree = shallow(<Sider />)
    expect(tree.find('MenuItem').length).toBe(2)
  })
})
