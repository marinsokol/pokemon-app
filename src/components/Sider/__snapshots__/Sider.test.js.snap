// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sider should render correctly 1`] = `
<Sider
  collapsedWidth={80}
  collapsible={true}
  defaultCollapsed={false}
  prefixCls="ant-layout-sider"
  reverseArrow={false}
  style={Object {}}
  width={200}
>
  <Menu
    className=""
    defaultSelectedKeys={
      Array [
        "1",
      ]
    }
    mode="inline"
    prefixCls="ant-menu"
    theme="dark"
  >
    <MenuItem
      key="1"
    >
      <Link
        replace={false}
        to="/"
      >
        <Icon
          type="heart-o"
        />
        <span>
          All pokemons
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="2"
    >
      <Link
        replace={false}
        to="/favourites"
      >
        <Icon
          type="star-o"
        />
        <span>
          Favourite pokemons
        </span>
      </Link>
    </MenuItem>
  </Menu>
</Sider>
`;
